src/pages/ImportData.test.js

```diff
@@ -32,25 +41,172 @@ describe('isProtected', () => {
     expect(protectedWarning).not.toBeNull();
     const noWarning = screen.queryByText('The dataset(s) you just chose to import to Terra will be made available to you', { exact: false });
     expect(noWarning).toBeNull();
   });
 
-  it.each(nonProtectedUrls)('should not render warning about protected data from %o', ({ url, format }) => {
-    render(h(ImportDataOverview, { header, snapshots, isDataset, snapshotResponses, url, format }));
+  it('should not render warning about unprotected data', () => {
+    render(h(ImportDataOverview, { header, snapshots, isDataset, snapshotResponses, url: 'https://google.com/file.pfb', isProtectedData: false }));
     const protectedWarning = screen.queryByText('The data you chose to import to Terra are identified as protected', { exact: false });
     expect(protectedWarning).toBeNull();
     const noWarning = screen.queryByText('The dataset(s) you just chose to import to Terra will be made available to you', { exact: false });
     expect(noWarning).not.toBeNull();
   });
 });
+
+jest.mock('src/components/workspace-utils', () => ({
+  ...jest.requireActual('src/components/workspace-utils'),
+  useWorkspaces: jest.fn().mockReturnValue({
+    loading: false,
+    workspaces: [
+      {
+        workspace: {
+          namespace: 'test-namespace',
+          name: 'protected-google',
+          cloudPlatform: 'Gcp',
+          googleProject: 'test-project-1',
+          workspaceId: 'ws-1',
+          bucketName: 'fc-secure-ws-1',
+        },
+        accessLevel: 'PROJECT_OWNER',
+      },
+      {
+        workspace: {
+          namespace: 'test-namespace',
+          name: 'unprotected-google',
+          cloudPlatform: 'Gcp',
+          googleProject: 'test-project-2',
+          workspaceId: 'ws-2',
+          bucketName: 'fc-ws-2',
+        },
+        accessLevel: 'OWNER',
+      },
+      {
+        workspace: { namespace: 'test-namespace', name: 'azure', cloudPlatform: 'Azure', workspaceId: 'ws-3' },
+        accessLevel: 'WRITER',
+      },
+    ],
+  }),
+}));
+
+describe('ImportDataDestination', () => {
+  it('should explain protected data restricts eligible workspaces', async () => {
+    render(
+      h(ImportDataDestination, {
+        workspaceId: null,
+        templateWorkspaces: [],
+        template: [],
+        userHasBillingProjects: true,
+        importMayTakeTime: true,
+        authorizationDomain: '',
+        onImport: () => {},
+        isImporting: false,
+        isProtectedData: true,
+      })
+    );
+    const existingWorkspace = screen.queryByText('Start with an existing workspace', { exact: false });
+    await userEvent.click(existingWorkspace); // select start with existing workspace
+
+    const protectedWarning = screen.queryByText('You may only import to workspaces with an Authorization Domain and/or protected data setting.', {
+      exact: false,
+    });
+    expect(protectedWarning).not.toBeNull();
```

Comment:
Elsewhere I see `.toBeTruthy()` instead of `.not.toBeNull()` (and `.toBeFalsy()` for `.toBeNull()`). Are the two the same? If so, is there a convention to use one or the other?

Reply:
I've seen both.  I think `toBeTruthy` covers null but also deals with something like empty strings instead of having to check all bases.  I *believe* in my situation null alone covers what I'm looking for here.